backend/portal/api/curation/v1/curation/collections/collection_id/datasets/dataset_id/actions.py

```diff
@@ -1,41 +1,163 @@
+
+    dataset = {
+        "assay": ontology_term_ids_to_response(d.metadata.assay),
+        "batch_condition": d.metadata.batch_condition,
+        "cell_count": d.metadata.cell_count,
+        "cell_type": ontology_term_ids_to_response(d.metadata.cell_type),
+        "dataset_assets": artifacts,
+        "development_stage": ontology_term_ids_to_response(d.metadata.development_stage),
+        "disease": ontology_term_ids_to_response(d.metadata.disease),
+        "donor_id": d.metadata.donor_id,
+        "explorer_url": "string",
+        "id": d.dataset_id.id,
+        "mean_genes_per_cell": d.metadata.mean_genes_per_cell,
+        "organism": ontology_term_ids_to_response(d.metadata.organism),
+        "processing_status": dataset_processing_status_to_response(d.status, d.dataset_id.id),
+        "processing_status_detail": "string",
+        "revised_at": "string", # TODO
+        "revision": 0,
+        "schema_version": d.metadata.schema_version,
+        "self_reported_ethnicity": ontology_term_ids_to_response(d.metadata.self_reported_ethnicity),
+        "sex": ontology_term_ids_to_response(d.metadata.sex),
+        "suspension_type": d.metadata.suspension_type,
+        "tissue": ontology_term_ids_to_response(d.metadata.tissue),
+        "x_approximate_distribution": d.metadata.x_approximate_distribution.upper(),
+    }
+
+    if d.status:
+        if d.status.processing_status == DatasetProcessingStatus.FAILURE:
+            if d.status.validation_status == DatasetValidationStatus.INVALID:
+                dataset["processing_status_detail"] = d.status.validation_message
+                dataset["processing_status"] = "VALIDATION_FAILURE"
+            else:
+                dataset["processing_status"] = "PIPELINE_FAILURE"
+        else:
+            dataset["processing_status"] = d.status.processing_status
+
+    dataset_ontology_elements = DATASET_ONTOLOGY_ELEMENTS_PREVIEW if preview else DATASET_ONTOLOGY_ELEMENTS
+    for ontology_element in dataset_ontology_elements:
+        if dataset_ontology_element := dataset.get(ontology_element):
+            if not isinstance(dataset_ontology_element, list):
+                # Package in array
+                dataset[ontology_element] = [dataset_ontology_element]
+        else:
+            dataset[ontology_element] = []
+
+    if not preview:  # Add these fields only to full (and not preview) Dataset metadata response
+        dataset["revision_of"] = d.canonical_dataset.dataset_id.id
+        dataset["title"] = d.metadata.name
+        if d.metadata.is_primary_data is not None:
+            dataset["is_primary_data"] = is_primary_data_mapping.get(d.metadata.is_primary_data, [])
+
+    return dataset
+
+
 def get(collection_id: str, dataset_id: str = None):
-    db_session = g.db_session
-    if not db_session.query(DbCollection.id).filter(DbCollection.id == collection_id).first():
+    business_logic = get_business_logic()
+
+    # TODO: double lookup
```

Comment:
Here we need to decide if we want to always do a double lookup or if those methods only make sense when using either the canonical or the version id. I can look at the notebooks and see what makes sense, or just add the double lookup anywhere.

Reply:
I added this function https://github.com/chanzuckerberg/single-cell-data-portal/blob/9e0a2cb7ecc9b009da4567670fc118cc4355fd7c/backend/portal/api/curation/v1/curation/collections/common.py#L287-L298
We can punt on the double look-up decision and be inefficient for now for the sake of time.

Reply:
I agree. I will use that function.